### ISSUE TITLE
Betts-Miller: the LCL-crossing layer takes a whole moist step, biasing the parcel warm

### ISSUE THREAD

**Issue by duncanwp** [climate-analytics-lab/jax-gcm#695]
Raised in review of #690 (Codex, P2). Follow-up to the parcel-water fix in that PR.

## What's wrong

`_parcel_ascent` in `jcm/physics/convection/betts_miller/betts_miller.py` steps the parcel level by level and picks either the dry-adiabatic or the moist-adiabatic candidate for each whole layer. A layer that **contains the LCL** cannot be represented by either branch, and the code's switch test is taken at the dry endpoint:

```python
t_dry  = theta0 * (p_k / c.p0) ** c.akap
newly_sat = q_parcel >= saturation_specific_humidity(t_dry, p_k)   # dry endpoint
...
t_moist = t_prev + _moist_dtdlnp(t_prev, q_prev) * log(p_k / p_prev)   # WHOLE layer
```

So at the crossing level the parcel banks a **full layer** of moist-lapse latent warming for a level it has only just reached saturation in, while `t_prev` was still on the dry adiabat. The state is internally inconsistent: `t_k` takes the moist candidate but `qsat(t_moist)` comes out *above* the water the parcel lifted, so the parcel is subsaturated at its own temperature.

This is not an edge case — it fires at every surface RH tested (0.60, 0.75, 0.85, 0.95).

## Evidence

Same ascent integrated on a 400x finer pressure grid (LCL resolved to ~0.06 hPa) as reference, 300 K surface parcel, 25 coarse levels:

| crossing rule | mean ΔT | max \|ΔT\| | water gained | convective onset on the scheme's own fixture |
|---|---|---|---|---|
| `q ≥ qsat(t_dry)`, `q = qsat(t_moist)` (pre-#690) | +0.96 K | 2.03 K | **+3.3 %** | RH 0.90 |
| `q ≥ qsat(t_dry)`, `q` capped at `q_parcel` (**current**) | +0.95 K | 2.01 K | 0.0 % | RH 0.90 |
| `q ≥ qsat(t_moist)` (defer the switch) | **−0.45 K** | **0.57 K** | 0.0 % | **RH 0.95** |

#690 landed the cap, which removes the invented water — the part that was unambiguously wrong — and leaves the temperature overshoot.

Deferring the switch was implemented and **not** landed: it is 3.5x more accurate in parcel temperature, but it errs *cold* (the level below the switch is left supersaturated on the dry branch, the opposite inconsistency) and that shifts convective onset on `_moist_unstable_column` from RH 0.90 to RH 0.95, turning three existing tests' columns non-convecting. Trading a measured behaviour change of that size for an accuracy gain is not something to do as a side-effect of an unrelated PR.

## Fix

Resolve the LCL **inside** the layer, which removes both inconsistencies and has no bias to trade:

1. Solve once for `p_lcl`, where the dry-adiabatic parcel saturates: `q_parcel = qsat(theta0·(p/p0)^κ, p)`. A few Newton iterations on `ln p`; it does not depend on the level, so it can be hoisted out of the scan.
2. In the crossing layer, integrate dry from `p_prev` to `p_lcl` (analytic — potential temperature is conserved), then moist from `(t_lcl, p_lcl)` to `p_k`.
3. `is_sat` becomes `p_k <= p_lcl`, so the switch test and the integration agree by construction and the `jnp.minimum(qsat_moist, q_parcel)` cap becomes provably non-binding.

Then re-check the three columns in `betts_miller_test.py` and re-derive the onset RH, since the reference profile `t_ref = tp` feeds the relaxation directly.

Isca's `betts_miller.f90` has the same whole-layer structure, so this is an improvement on the reference rather than a fidelity gap against it — worth stating explicitly in whatever lands.

## Refs

Cross-referenced from the comment at the `q_k` site. Related: #661 (the parcel-water class this came from), #694 (the other live instance, the downdraft wet-bulb).
